backend-shared/util/operations/types_test.go

```diff
@@ -102,6 +103,167 @@ var _ = Describe("Testing CreateOperation function.", func() {
 			Expect(dbOperationFirst.SeqID).To(Equal(dbOperationSecond.SeqID))
 		})
 	})
+
+	Context("Testing waitForOperationToComplete and IsOperationComplete function.", func() {
+		var ctx context.Context
+		var dbq db.AllDatabaseQueries
+		var k8sClient *sharedutil.ProxyClient
+		var logger logr.Logger
+
+		delayUpdateOfState := func(operation *db.Operation, k8sClient client.Client, gitopsEngineInstance db.GitopsEngineInstance) {
+			Eventually(func() bool {
+				// Allow waitForOperationToComplete to iterate/wait
+				time.Sleep(2 * time.Second)
```

Comment:
Would it be possible to remove the `time.Sleep`? I ask  because Sleep statements in test make me very wary (as a source for race conditions)...  I checked the logic of the test and I THINK this `time.Sleep` isn't actually doing anything, so it should be easy to remove.

Reply:
The purpose of the delay is to give time for the for loop in `waitForOperationToComplete` in types.go to iterate a few times. Removing the delay does seems to work ok, and the for loop gets executed once.

Try adding a Println just inside the for loop and try adjusting the delay to 2-5 seconds.  Your println statement will get executed more than once.  Thus, IsOperationComplete also gets called multiple times.

Instead of sleep, I've tried the following, and it simulates the delay too.

```
				currentTime := time.Now()
				for time.Now().Before(currentTime.Add(2 * time.Second)) {
				}
```

If you prefer, we can just remove this.

Reply:
I believe you can get rid of the delay as long as you convert the Expects to return bools, as above.

Reply:
See my other comment. https://github.com/redhat-appstudio/managed-gitops/pull/614#discussion_r1313213676